lib/core/connector_settings.rb

```diff
@@ -114,6 +114,16 @@ def valid_index_name?
       index_name&.start_with?(Utility::Constants::CONTENT_INDEX_PREFIX)
     end
 
+    def ready_for_sync?
+      Connectors::REGISTRY.registered?(service_type) &&
+        valid_index_name? &&
+        connector_status_allows_sync?
+    end
+
+    def running?
+      @elasticsearch_response[:_source][:last_sync_status] == Connectors::SyncStatus::IN_PROGRESS
```

Comment:
I'm OK with this now, but in long term, we probably should not rely on this, but on `.elastic-connectors-sync-jobs`.